### PR TITLE
Parallelize the data-dependencies test.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
It runs about twice as fast on my laptop (240s -> 120s) when allowed to
run tests in parallel. I'm not sure this will work or make a huge
difference when it comes to timeouts in CI, but it's worth a shot.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/tests/src/DA/Test/DataDependencies.hs [hunk 1]
```diff
@@ -27,7 +26,6 @@ import SdkVersion
 
 main :: IO ()
 main = do
-    setEnv "TASTY_NUM_THREADS" "1" True
```
Comment (cocreature) [inline compiler/damlc/tests/src/DA/Test/DataDependencies.hs:30]: given that you report a 2x speedup how much do we gain by setting this to 2 and 4? We should also change the resources Bazel allocates for this either if we don’t fix the number of CPUs here by marking it exclusive or via `    tags = ["cpu:2"],`.

Comment (sofiafaro-da) [reply compiler/damlc/tests/src/DA/Test/DataDependencies.hs:30]: I just tried this now on my laptop:

- Setting TASTY_NUM_THREADS to 1: 229.0s
- Setting TASTY_NUM_THREADS to 2: 142.3s
- Setting TASTY_NUM_THREADS to 3: 103.9s
- Setting TASTY_NUM_THREADS to 4: 96.5s
- Not setting TASTY_NUM_THREADS: 134.5s

I'm going to set it to 3. Should I change anything else (e.g. in Bazel)?

Comment (cocreature) [reply compiler/damlc/tests/src/DA/Test/DataDependencies.hs:30]: sounds good, then let’s add `cpu:3` to the tags to make sure we don’t schedule too much.

**Review by cocreature — APPROVED** [review]
Thanks!

**Pull request merged**